### PR TITLE
Associations 

### PR DESCRIPTION
**Pull request by Chimoneg27**
### For this feature I did the following

- Made sure to remove linter errors
- Used git flow
- Created associations between associations between classes
- Created the book, rental and classroom class
- Used ruby best practices

### CODE REVIEW

File: class/book.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rental = []
+  end
+
+  def add_rental(date, person)
+    Rental.new(date, person, self)
+  end
```
Comment (danmainah) [inline class/book.rb:12]: - [ ] @Chimoneg27 Great work implementing the `has-many ` side of the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental, however you forgot to do the same in your person class, kindly consider implementing it to achieve the many-to-many relationship

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Chimoneg27 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Implemented the has-many/belongs-to relationship between Classroom and Student.
✔️  Implemented the belongs-to relationship( for the many-to-many relationship between Person and Book) in the Rental class.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: class/book.rb (hunk 1 above)
Comment (Abenezer-Tilahun) [reply class/book.rb:12]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @Chimoneg27 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.


## Optional suggestions 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Pull request merged**